cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+	"syscall"
+
+	"google.golang.org/grpc/status"
+
+	pb "github.com/GoogleCloudPlatform/ops-agent/cmd/ops_agent_uap_plugin/google_guest_agent/plugin"
+)
+
+const (
+	OpsAgentConfigLocationLinux = "/etc/google-cloud-ops-agent/config.yaml"
+	ConfGeneratorBinary         = "libexec/google_cloud_ops_agent_engine"
+	LogsDirectory               = "log/google-cloud-ops-agent"
+	FluentBitStateDiectory      = "state/fluent-bit"
+	FluentBitRuntimeDirectory   = "run/google-cloud-ops-agent-fluent-bit"
+	OtelRuntimeDirectory        = "run/google-cloud-ops-agent-opentelemetry-collector"
+	DefaultPluginStateDirectory = "/var/lib/google-guest-agent/plugins/ops-agent-plugin"
+)
+
+var (
+	AgentServiceNameRegex    = regexp.MustCompile(`[\w-]+\.service`)
+	AgentSystemdServiceNames = []string{"google-cloud-ops-agent.service", "stackdriver-agent.service", "google-fluentd.service"}
+)
+
+// Apply applies the config sent or performs the work defined in the message.
+// ApplyRequest is opaque to the agent and is expected to be well known contract
+// between Plugin and the server itself. For e.g. service might want to update
+// plugin config to enable/disable feature here plugins can react to such requests.
+func (ps *OpsAgentPluginServer) Apply(ctx context.Context, msg *pb.ApplyRequest) (*pb.ApplyResponse, error) {
+	return &pb.ApplyResponse{}, nil
+}
+
+// Start starts the plugin and initiates the plugin functionality.
+// Until plugin receives Start request plugin is expected to be not functioning
+// and just listening on the address handed off waiting for the request.
+func (ps *OpsAgentPluginServer) Start(ctx context.Context, msg *pb.StartRequest) (*pb.StartResponse, error) {
+	if ps.cancel != nil {
+		log.Printf("The Ops Agent plugin is started already, skipping the current request")
+		return &pb.StartResponse{}, nil
+	}
+	log.Printf("Received a Start request: %s. Starting the Ops Agent", msg)
+
+	pContext, cancel := context.WithCancel(context.Background())
+	ps.cancel = cancel
+
+	pluginStateDir := msg.GetConfig().GetStateDirectoryPath()
+	if pluginStateDir == "" {
```

Comment:
Under what conditions will this be empty? If empty is unexpected, we should return an error. On the other hand, if we're required to handle empty, then we should use a default state path that _we_ own, not one that the plugin owns.